nUpdate.WPFUserInterface/UpdaterUI.cs

```diff
@@ -0,0 +1,120 @@
+                {
+                    var vmUpdateSearch = new WPFUserInterface.ViewModel.UpdateSearchVm(UpdateManager);
+                    dialogService.ShowDialog("searchforupdates", vmUpdateSearch);
+
+                    if (vmUpdateSearch.HasError) return;
+
+                    if (!vmUpdateSearch.UpdatesFound)
+                    {
+                        messageboxService.Show(lp.NoUpdateDialogInfoText, lp.NoUpdateDialogHeader, EnuMessageBoxButton.Ok,
+                            EnuMessageBoxImage.Information);
+                        return;
+                    }
+                }
+                else
+                {
+                    try
+                    {
+                        if (!await UpdateManager.SearchForUpdatesAsync()) return;
+                    }
+                    catch (Exception ex)
+                    {
+                        messageboxService.Show(ex.Message, lp.UpdateSearchErrorCaption, EnuMessageBoxButton.Ok,
+                            EnuMessageBoxImage.Error);
+                        return;
+                    }
+                }
+
+                var vmChangelog = new WPFUserInterface.ViewModel.ChangelogVm(UpdateManager);
+                dialogService.ShowDialog("showChangelog", vmChangelog);
+
+                if (!vmChangelog.DialogResult) return;
+
+                var vmDownload = new WPFUserInterface.ViewModel.DownloadUpdateVm(UpdateManager);
+                dialogService.ShowDialog("downloadUpdates", vmDownload);
+
+                if (!vmDownload.DialogResult) return;
+
+                bool valid;
+                try
+                {
+                    valid = UpdateManager.ValidatePackages();
+                }
+                catch (FileNotFoundException)
+                {
+                    messageboxService.Show(lp.PackageNotFoundErrorText, lp.PackageValidityCheckErrorCaption, EnuMessageBoxButton.Ok,
+                        EnuMessageBoxImage.Error);
+                    return;
+                }
+                catch (ArgumentException)
+                {
+                    messageboxService.Show(lp.InvalidSignatureErrorText, lp.PackageValidityCheckErrorCaption, EnuMessageBoxButton.Ok,
+                        EnuMessageBoxImage.Error);
+                    return;
+                }
+                catch (Exception ex)
+                {
+                    messageboxService.Show(ex.Message, lp.PackageValidityCheckErrorCaption, EnuMessageBoxButton.Ok,
+                        EnuMessageBoxImage.Error);
+                    return;
+                }
+
+                if (!valid)
+                {
+                    messageboxService.Show(lp.SignatureNotMatchingErrorText, lp.InvalidSignatureErrorCaption, EnuMessageBoxButton.Ok,
+                        EnuMessageBoxImage.Error);
+                }
+
+                else
+                    try
+                    {
+                        UpdateManager.InstallPackage();
```

Comment:
Where is this method implemented in a custom manner? This would call the default update installer (Windows Forms), wouldn't it?

Reply:
No, because i must implement my own UpdateManager, same mistake as the other complaints i think.